File: win-prog/ProgWinVB/Brushes and Pens/GradientPen/GradientPen.vb
'--------------------------------------------
' GradientPen.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Class GradientPen
    Inherits PrintableForm

    Shared Shadows Sub Main()
        Application.Run(New GradientPen())
    End Sub

    Sub New()
        Text = "Gradient Pen"
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        Dim lgbr As New LinearGradientBrush( _
                                New Rectangle(0, 0, cx, cy), _
                                Color.White, Color.Black, _
                                LinearGradientMode.BackwardDiagonal)
        Dim pn As New Pen(lgbr, Math.Min(cx, cy) \ 25)
        pn.Alignment = PenAlignment.Inset

        grfx.DrawRectangle(pn, 0, 0, cx, cy)
        grfx.DrawLine(pn, 0, 0, cx, cy)
        grfx.DrawLine(pn, 0, cy, cx, 0)
    End Sub
End Class

File: win-prog/ProgWinCS/Brushes and Pens/GradientPen/GradientPen.cs
//------------------------------------------
// GradientPen.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

class GradientPen: PrintableForm 
{
     public new static void Main()
     {
          Application.Run(new GradientPen());
     }
     public GradientPen()
     {
          Text = "Gradient Pen";
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          Brush lgbrush = new LinearGradientBrush(
                                   new Rectangle(0, 0, cx, cy), 
                                   Color.White, Color.Black,
                                   LinearGradientMode.BackwardDiagonal);

          Pen pen = new Pen(lgbrush, Math.Min(cx, cy) / 25);

          pen.Alignment = PenAlignment.Inset;

          grfx.DrawRectangle(pen, 0, 0, cx, cy);
          grfx.DrawLine(pen, 0, 0, cx, cy);
          grfx.DrawLine(pen, 0, cy, cx, 0);
     }
}
